# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: docshell/base/BrowsingContext.h
# Commit: a8ac7bcafc27
# Full Hash: a8ac7bcafc270b5be3eca84efebcde672fe8fba5
# Author: Kris Maglione <maglione.k@gmail.com>
# Date: 2019-08-15 09:46:49
# Regressor Bug: 1562292
# File Overlap Count: 1
# Description:
#   Bug 1562292: Part 1b.2 - Don't fall back to named lookup for inaccessible special name. r=nika
#   
#   Currently, if a window with a special name is inaccessible to the caller, we
#   fall back to ordinary named lookup, which is not desirable.
#   
# ==============================================================================

diff -r f1fc2382346d -r a8ac7bcafc27 docshell/base/BrowsingContext.h
--- a/docshell/base/BrowsingContext.h	Thu Aug 01 18:45:47 2019 -0700
+++ b/docshell/base/BrowsingContext.h	Thu Aug 08 11:43:38 2019 -0700
@@ -394,6 +394,10 @@
                   uint64_t aBrowsingContextId, Type aType);
 
  private:
+  // Returns true if the given name is one of the "special" names, currently:
+  // "_self", "_parent", "_top", or "_blank".
+  static bool IsSpecialName(const nsAString& aName);
+
   // Find the special browsing context if aName is '_self', '_parent',
   // '_top', but not '_blank'. The latter is handled in FindWithName
   BrowsingContext* FindWithSpecialName(const nsAString& aName,
